src/Symfony/Component/Security/Core/User/UserInterface.php

```diff
@@ -61,6 +63,8 @@ public function getPassword();
      *
      * This can return null if the password was not hashed using a salt.
      *
+     * This method is deprecated since Symfony 5.3, implement it from {@link LegacyPasswordAuthenticatedUserInterface} instead.
```

Comment:
```suggestion
     * @deprecated since Symfony 5.3, implement it from {@link LegacyPasswordAuthenticatedUserInterface} instead.
```

Reply:
This is on purpose: using `@deprecated` means we ask developers to stop implementing the method in their user class, while we just want it to implement another interface with the very same method.